addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+                    PercentType vol = readVolume();
+                    updateState(CHANNEL_VOLUME, vol);
+                } else if ("mute".equals(command)) {
+                    String muteState = readUntil('!');
+                    updateState(CHANNEL_MUTE, "on".equals(muteState) ? OnOffType.ON : OnOffType.OFF);
+                } else if ("power_off".equals(command)) {
+                    power = false;
+                    updateState(CHANNEL_MUTE, OnOffType.OFF);
+                    updateState(CHANNEL_POWER, OnOffType.OFF);
+                } else if ("power_on".equals(command)) {
+                    power = true;
+                    updateState(CHANNEL_POWER, OnOffType.ON);
+                    powerOnRefresh();
+                } else if ("power".equals(command)) {
+                    String state = readUntil('!');
+                    if ("on".equals(state)) {
+                        power = true;
+                        updateState(CHANNEL_POWER, OnOffType.ON);
+                        powerOnRefresh();
+                    } else if ("standby".equals(state)) {
+                        updateState(CHANNEL_MUTE, OnOffType.OFF);
+                        updateState(CHANNEL_POWER, OnOffType.OFF);
+                        power = false;
+                    }
+                } else if ("dimmer".equals(command)) {
+                    updateState(CHANNEL_BRIGHTNESS, readDimmer());
+                } else if ("freq".equals(command)) {
+                    updateState(CHANNEL_FREQUENCY, readFrequency());
+                } else if ("source".equals(command)) {
+                    updateState(CHANNEL_SOURCE, new StringType(readUntil('!')));
+                } else if ("display".equals(command)) {
+                    String stringLength = readUntil(',');
+                    int length = Integer.parseInt(stringLength);
+                    byte[] data = new byte[length];
+                    int off = 0;
+                    while (off != length) {
+                        int r = serialPort.getInputStream().read(data, off, length - off);
+                        if (r == -1) {
+                            throw new IOException("Connection closed while reading display content");
+                        } else {
+                            off += r;
+                        }
+                    }
+                    // We don't do anything with display content, could add a channel if this is useful
+                } else {
+                    readUntil('!'); // discard
+                }
+
+            } catch (IOException e) {
+                if (serialPort != null) {
+                    logger.debug("Input error while receiving data from amplifier, waiting...", e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    disconnect();
+                    try {
+                        Thread.sleep(ERROR_RETRY_DELAY_MS);
```

Comment:
Instead of the sleep you probably can also use the schedule with delay. So when this thread gets the exception it will be closed, but restarted with the delay. This will give the same effect. If you do want to use sleep and catch the InterruptedException you need to handle it correctly and set the interrupted flag again: `Thread.currentThread().interrupt();`.